src/ripple/app/misc/impl/Manifest.cpp

```diff
@@ -496,10 +512,11 @@ ManifestCache::applyManifest(Manifest m)
             m.sequence,
             iter->second.sequence);
 
-    signingToMasterKeys_.erase(iter->second.signingKey);
+    assert(iter->second.signingKey || iter->second.revoked());
+    signingToMasterKeys_.erase(*iter->second.signingKey);
```

Comment:
It should be impossible for a revoked key to get to this point, so the assert should be `assert(iter->second.signingKey && !iter->second.revoked())`.

Reply:
I'm trying to better understand this piece of code. I found that `assert(m == iter->second)` fails. Specifically, `m` has a higher sequence number than the manifest in `iter->second`. Isn't that wrong? I was of the understanding that `map_` (parent data structure of `iter`) needs to be identical to the input of the function (`m`)

Reply:
`iter->second` is holding the old / existing manifest. `m` is the new manifest. They're for the same `masterKey`, but they're expected to be different. In fact, `prewriteCheck` checks that.

You could assert that 
```
assert(
        iter->second.masterKey == m.masterKey &&
        iter->second.sequence < m.sequence);
```
but that's redundant.